### DISCUSSION TITLE
esp8226 wiring for INMP441

### DISCUSSION THREAD

**Discussion by PointlessOtter** [pschatzmann/arduino-audio-tools#453] [Q&A] [answered]
Hi,
I am not understanding how could I use my esp8226 with INMP441 to stream audio. I did found the wiring diagram for esp32 (https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-webserver/streams-i2s-webserver_wav) but I am struggling on how to wire up esp8226 correctly.
Can you please pin-point me in the right direction to figure this out? Thanks!

**Comment by pschatzmann** [comment]
I am confused. I never heard of an esp8226 ?

**Reply by PointlessOtter** [reply]
Sorry, mistyped :/ - **ESP8266**

**Comment by pschatzmann** [comment]
The ESP8266 pins for I2S are:
Data bits (SD) = GPIO3/RX0
Data bit-clock (SCK) = GPIO15
Word select (WS) = GPIO2/TX1

I have never tested this however...

**Reply by PointlessOtter** [reply]
Ah, too bad. Good news, that did already order an ESP32 👍 
Still, tried with your suggestion and hooked up ESP8266 to INMP441:
|INMP441|ESP8266 |
|-|-|
|(SD) | GPIO3/RX0|
|(SCK) | GPIO15|
|(WS) | GPIO2/TX1|
|(L/R) | GND|
|(VDD) | 3V|
|(GND) | GND|

Tried running the **stream-i2s-serial** but got an an error :/

![image](https://user-images.githubusercontent.com/39235428/202487109-d935bf4d-941b-49e0-b68d-da5f9fb7d66d.png)

**Comment by pschatzmann** [comment]
These options are only supported by the ESP32. 
Therefore  you need to omit time in you sketch...

**Reply by PointlessOtter** [reply]
Thanks, omitted the two settings and was able to upload the sketch.
```
    //cfg.use_apll = false;  // try with yes
    //cfg.pin_mck = 3; 
```
Unfortunately no success in reading the INMP441 
![image](https://user-images.githubusercontent.com/39235428/202524981-03f89291-5fbe-46b9-bbb7-ef657aa374a4.png)


Do I by any chance actually need to define at top which pins are for what?

**Comment by pschatzmann** [comment, marked as answer]
I found the following in the Arduino source code:
// IOs used for I2S. Not defined in i2s.h, unfortunately.
// Note these are internal GPIO numbers and not pins on an
// Arduino board. Users need to verify their particular wiring.
#define I2SO_DATA 3
#define I2SO_BCK  15
#define I2SO_WS   2
#define I2SI_DATA 12
#define I2SI_BCK  13
#define I2SI_WS   14

I suggest that you try the I pins for input and the O for output

**Reply by PointlessOtter** [reply]
Tried connecting like this but unfortunately with no success :(
|INMP441|	ESP8266|
|-|-|
|(SD)	|GPIO12|
|(SCK)	|GPIO15|
|(WS)	|GPIO2|
|(L/R)|	GND|
|(VDD)|	3V|
|(GND)|	GND|

**Reply by pschatzmann** [reply]
sck should be 13!

**Reply by PointlessOtter** [reply]
Unfortunately have no success :( Anyway, much appreciate your time and help - will get my ESP32 and should be much straightforward! :)
